src/VirtualClient/VirtualClient.Dependencies/NvidiaContainerToolkitInstallation.cs

```diff
@@ -183,29 +188,55 @@ private List<string> NvidiaContainerToolkitInstallationCommands(LinuxDistributio
             return commands;
         }
 
-        private Task ExecuteCommandAsync(string commandLine, string workingDirectory, EventContext telemetryContext, CancellationToken cancellationToken)
+        private async Task<bool> CheckIfNvidiaContainerToolkitInstalledAsync(LinuxDistribution linuxDistribution, EventContext telemetryContext, CancellationToken cancellationToken)
         {
-            EventContext relatedContext = telemetryContext.Clone();
+            string unixCommand = string.Empty;
 
-            return this.RetryPolicy.ExecuteAsync(async () =>
+            switch (linuxDistribution)
             {
-                using (IProcessProxy process = this.systemManager.ProcessManager.CreateElevatedProcess(this.Platform, commandLine, null, workingDirectory))
-                {
-                    this.CleanupTasks.Add(() => process.SafeKill());
-                    this.LogProcessTrace(process);
+                case LinuxDistribution.Ubuntu:
+                    unixCommand = $"dpkg -l | grep -i \"{NvidiaContainerToolkit}\"";
+                    break;
+                case LinuxDistribution.Debian:
+                    unixCommand = $"dpkg -l | grep \"{NvidiaContainerToolkit}\"";
+                    break;
+                case LinuxDistribution.CentOS8:
+                case LinuxDistribution.CentOS7:
+                case LinuxDistribution.RHEL7:
+                    unixCommand = $"yum list installed | \"grep {NvidiaContainerToolkit}\"";
+                    break;
+                case LinuxDistribution.SUSE:
+                    unixCommand = $"zypper se -i | \"grep {NvidiaContainerToolkit}\"";
+                    break;
 
-                    await process.StartAndWaitAsync(cancellationToken)
-                        .ConfigureAwait(false);
+                default:
+                    return false;
+            }
 
-                    if (!cancellationToken.IsCancellationRequested)
-                    {
-                        await this.LogProcessDetailsAsync(process, relatedContext, "NvidiaToolkitInstallation")
-                            .ConfigureAwait(false);
+            string executeCommand = $"bash -c \"{unixCommand}\"";
 
-                        process.ThrowIfErrored<DependencyException>(errorReason: ErrorReason.DependencyInstallationFailed);
-                    }
+            var process = await this.ExecuteCommandAsync(executeCommand, null, Environment.CurrentDirectory, telemetryContext, cancellationToken, true)
+                .ConfigureAwait(false);
+
+            if (ProcessProxy.DefaultSuccessCodes.Contains(process.ExitCode))
+            {
+                string output = process.StandardOutput.ToString();
+
+                string pattern = @"nvidia-container-toolkit\s+(\S+)";
```

Comment:
again, I would suggest we just dump the output of `grep` to kusto and write kusto queries to parse. We'll have these kusto queries checked in to the same repo as VC